sdks/java/io/kafka/src/main/java/org/apache/beam/sdk/io/kafka/KafkaSinkMetrics.java

```diff
@@ -97,7 +86,15 @@ public static Gauge createBacklogGauge(String topic, int partitionId) {
    * @return Counter.
    */
   public static Gauge createBacklogGauge(MetricName name) {
-    return new DelegatingGauge(name, false, true);
+    // Use label to differenciate between the type of gauge metric is created
+    // TODO(34195):
+    if (name.getLabels().containsKey(MonitoringInfoConstants.Labels.PER_WORKER_METRIC)
```

Comment:
maybe we could add a method for this in MonitoringInfoConstant? woudl reduce duplication of this lookup/comparison logic:

boolean isPerWorkerMetric(Map labels)